lib/practice_exercises.rb

```diff
@@ -1,13 +1,47 @@
+#!/usr/bin/ruby
+# 
+# Title  : Array Strings - Ada Cohort 13 - Space
+# Author : Suely Barreto
+# Date   : March 2020
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Given a string, determine if it is a palindrome considering only alphanumeric characters and ignoring cases
+# Note: For the purpose of this problem, we define empty string as valid palindrome.
+
+# Time complexity : O(n)
+# Space complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^a-z]/, '')
+
+  i = 0
+  j = string.length - 1
+  while i < j do
+    return false if string[i] != string[j]
+    i += 1
+    j -= 1
+  end
+  return true
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity : O(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word.  Otherwise 👍